crates/storage/src/types.rs

```diff
@@ -28,7 +28,9 @@ pub struct StateUpdate {
     /// None for `pending`
     #[serde(default)]
     pub block_hash: Option<StarknetBlockHash>,
-    pub new_root: StateCommitment,
+    /// None for `pending`
+    #[serde(default)]
+    pub new_root: Option<StateCommitment>,
```

Comment:
This is probably the simplest way to implement this, _but_: pending state updates don't make it to the DB, so it would be great if we could keep `new_root` non-optional here. That way it would be explicit that state updates read from the storage _do_ have `new_root`...

Reply:
Yep, the `new_root` is `required` according to the spec: https://github.com/starkware-libs/starknet-specs/blob/master/api/starknet_api_openrpc.json#L917

Reply:
Nice catch (in storage). Will fix it.